### ISSUE TITLE
BoxGroupIterator.hasNext() causes NullPointerException

### ISSUE THREAD

**Issue by sohtsuka** [box/box-java-sdk#98]
Calling `BoxGroupIterator.hasNext() ` causes NPE. Stacktrace:

```
Exception in thread "main" java.lang.NullPointerException
	at com.box.sdk.JSONIterator.loadNextJsonObject(JSONIterator.java:96)
	at com.box.sdk.JSONIterator.hasNext(JSONIterator.java:32)
	at com.box.sdk.BoxGroupIterator.hasNext(BoxGroupIterator.java:20)
```

**Comment by sohtsuka** [comment]
I have fix this up on sohtsuka@ea2d066be815aa4be2c07e6482dd31f1a39d8b5d . If this change is suitable for the SDK, can I make a pull request?

**Comment by gcurtis** [comment]
Thanks for the fix! If you submit that change as a pull request I'll be sure to merge it.
